core/src/main/scala/kafka/controller/PartitionStateMachine.scala

```diff
@@ -146,18 +146,17 @@ class PartitionStateMachine(controller: KafkaController) extends Logging {
     val topicAndPartition = TopicAndPartition(topic, partition)
     val currState = partitionState.getOrElseUpdate(topicAndPartition, NonExistentPartition)
     try {
+      assertValidTransition(topicAndPartition, targetState)
       targetState match {
         case NewPartition =>
           // pre: partition did not exist before this
```

Comment:
If the pre comments just repeat what's in the code (which was the case for the ones I checked), I'd personally remove them. I saw the post comments as well. Since this PR is only about the valid transition states, I'd be OK with keeping things as they are for the post comments.